advocacy_docs/supported-open-source/barman/pg-backup-api/03-tasks.mdx

```diff
@@ -20,68 +20,224 @@ tags:
+    You need to set the `content-type` header as `application/json` when executing `POST` requests to this endpoint, otherwise you will receive a `400 Bad Request error`.
+
+##### recovery
+
+This operation is used to request a remote execution of `barman recover` command.
+
+The JSON payload of the request must contain the following keys:
+
+ * `type`: set as `recovery` to request that operation;
+ * `backup_id`: ID of the backup to be restored from Barman server `NAME`. Supports whatever `barman recover` supports, including `latest` for example;
+ * `remote_ssh_command`: the SSH command to be used to connect to the target Postgres server where the backup will be restored;
+ * `destination_directory`: the path in the target server where the backup will be restored.
 
+As an example, you could request pg-backup-api to restore the latest backup of Barman server `db_server_two` with a payload like this:
+
+```json
+{
+  "type": "recovery",
+  "backup_id": "latest",
+  "remote_ssh_command": "ssh postgres@db_server_two.my.org",
+  "destination_directory": "/var/lib/pgdata"
+}
+```
 
 ```bash
-curl -X POST http://barman-server.my.org/servers/db_server_two/operations -H "content-type: application/json" -d@payload-pg-backup-api.json
+curl -X POST http://barman-server.my.org/servers/db_server_two/operations -H "content-type: application/json" -d @payload-pg-backup-api.json
 {
-  "operation_id": {  "20230223T093201" },
+  "operation_id": "20240118T140951"
 }
 ```
 
-In the response above, "20230223T093201" is the OPERATION_ID which we will use next to check its status.
+The `operation_id` in the response is the ID of the operation that has been created in by the API.
 
+##### config_switch
 
-#### GET: /servers/NAME/operations/OPERATION_ID
+This operation is used to request a remote execution of `barman config-switch` command.
+
+The JSON payload of the request must contain the following key:
+
+ * `type`: set as `config_switch` to request that operation.
+
+Besides that, it may contain either of the following keys:
 
-This method allows you to check if an operation is: DONE, SUCCESS or IN_PROGRESS.
+ * `model_name`: the name of a model, if you want to apply it to the Barman server `NAME`; or
+ * `reset`: set with `true` if you want to unapply the currently active model of the Barman server `NAME`
+
+As an example, you could request pg-backup-api to apply the model `my-model` to the Barman server with a payload like this:
+
+```json
+{
+  "type": "config_switch",
+  "model_name": "my-model"
+}
+```
 
 ```bash
-curl http://barman-server.my.org/servers/db_server_two/operations/20230223T093201
+curl -X POST http://barman-server.my.org/servers/db_server_two/operations -H "content-type: application/json" -d @payload-pg-backup-api.json
 {
-  "recovery_id": "20230223T093201",
-  "status": "DONE"
+  "operation_id": "20240118T141135"
 }
 ```
+
+The `operation_id` in the response is the ID of the operation that has been created in by the API.
```

Comment:
I think we need a comma "The `operation_id` in the **response,** is the ID "

Reply:
I would use a comma if the sentence was like this:

"The operation_id in the response, `it` is the ID"

I'm not sure if a comma applies without an `it` or anything else to take a "break" in the sentence.

What do you think?

And what do you think @martinmarques ? You are way better than me on formal writing 😆

Reply:
Yeah, we don't need "it" in this case because the phrase is focusing on describing the ID

Reply:
We should try to avoid using "it" as much as possible

Reply:
No need for a coma there